server/controllers/journal/cash.js

```diff
@@ -0,0 +1,411 @@
+var q = require('q'),
```

Comment:
You put cash and cash discard logic in the same controller but why not for sale and sale discard? and also caution should be in the cash controller like cash and cash discard according to your logic.